tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -90,10 +88,17 @@ int IMessagePackFormatter<TraceChunkModel>.Serialize(ref byte[] bytes, int offse
         // overload of IMessagePackFormatter<TraceChunkModel>.Serialize() with `in` modifier on `TraceChunkModel` parameter
         public int Serialize(ref byte[] bytes, int offset, in TraceChunkModel traceChunk, IFormatterResolver formatterResolver, int? maxSize = null)
         {
+            if (traceChunk.SpanCount > 0)
+            {
+                var spanModel = traceChunk.GetSpanModel(0);
+                traceChunk.Tags?.FixTraceIdTag(spanModel.Span.TraceId128);
```

Comment:
We are intentionally adding 128-bit support for traces only for now. (And for logs injection but disabled by default.) Support for other products may come later.

(side note: DBM got partial support automatically because they re-use the W3C headers, although I don't know if they will use the full trace id or truncate it to 64-bit.)